pangeo_forge_recipes/aggregation.py

```diff
@@ -287,13 +287,19 @@ def schema_to_zarr(
     attrs: Optional[Dict[str, str]] = None,
     consolidated_metadata: Optional[bool] = True,
     encoding: Optional[Dict] = None,
+    append_dim: Optional[str] = None,
 ) -> zarr.storage.FSStore:
     """Initialize a zarr group based on a schema."""
+    if append_dim:
+        # if appending, only keep schema for coordinate to append. if we don't drop other
+        # coords, we may end up overwriting existing data on the `ds.to_zarr` call below.
+        schema["coords"] = {k: v for k, v in schema["coords"].items() if k == append_dim}
```

Comment:
Just realizing that this, while seemingly necessary to avoid overwriting non-`append_dim` coords, may prevent us from benefiting from the dimension consistency checks offered by `ds.to_zarr`... we may need to re-implement such checks in `StoreToZarr.__post_init__` if we want them.

Reply:
Thinking about this further... in `StoreToZarr.__post_init__` we actually don't know what the coordinates in the new data are (aside from the user-provided `append_dim`), because we haven't actually fetched and opened any source files yet.

So if we want to check for consistency, we could actually do it right here. Before dropping non-append coords from the schema, we would just need to open the target dataset and check that all coords match first. Could be a good idea.

Reply:
Having just reviewed [xarray's checks](https://github.com/pydata/xarray/blob/6af547cdd9beac3b18420ccb204f801603e11519/xarray/backends/api.py#L1828-L1842) again, I actually don't see that this is done there, so may be overly strict. I think we can move forward with what we have for now.